### PR TITLE
Brussels | ITP-2026-1 | Meysam Razagh | Sprint-3 | Practice TDD

### PR DESCRIPTION
**Pull request by Epunch**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist
In this PR, I practiced Test-Driven Development (TDD) by implementing three functions with their corresponding Jest tests:
* `countChar`: A function that counts occurrences of a character in a string.
* `repeat`: A function that repeats a string a specified number of times, including handling edge cases like 0 and negative counts.
* `getOrdinalNumber`: A function that converts numbers to their ordinal format (1st, 2nd, 3rd, etc.).
* All logic was developed following the Red-Green-Refactor cycle, ensuring 100% test coverage for the required scenarios.
## Questions

No questions, everything went smoothly!

### CODE REVIEW

File: Sprint-3/2-practice-tdd/get-ordinal-number.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+  }
+  if (num === 2) {
+    return "2nd";
+  }
+  if (num === 3) {
+    return "3rd";
+  }
+
+  return num + "th";
 }
```
Comment (noahg9) [inline Sprint-3/2-practice-tdd/get-ordinal-number.js:15]: This works for all numbers up to 20. Even better would be if you can make it work for all numbers. Right now 21 would output 21th instead of 21st and so on

File: Sprint-3/2-practice-tdd/get-ordinal-number.test.js [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+// Case 3: Identify the ordinal number for 3
+test("should return '3rd' for 3", () => {
+  expect(getOrdinalNumber(3)).toEqual("3rd");
+});
+
+// Case 4: Handle other numbers
+test("should return 'th' for numbers other than 1, 2, or 3", () => {
+  expect(getOrdinalNumber(4)).toEqual("4th");
+  expect(getOrdinalNumber(10)).toEqual("10th");
+});
```
Comment (noahg9) [inline Sprint-3/2-practice-tdd/get-ordinal-number.test.js:26]: Add tests for numbers ending in 1, 2 or 3 after 20. They will fail with your current implementation

Comment (Epunch) [reply Sprint-3/2-practice-tdd/get-ordinal-number.test.js:26]: @noahg9  I've added the test cases for numbers ending in 1, 2, and 3 after 20 as requested. They are now passing with the updated logic in the main file.

File: Sprint-3/2-practice-tdd/get-ordinal-number.js (hunk 1 above)
Comment (Epunch) [reply Sprint-3/2-practice-tdd/get-ordinal-number.js:15]: @noahg9 I've refactored the function using the modulo operator to ensure it's scalable for all numbers. It now passes the new test cases. Thanks for the feedback!

Comment (noahg9) [reply Sprint-3/2-practice-tdd/get-ordinal-number.js:15]: Perfect!

**Review by noahg9 — APPROVED** [review]
